### DISCUSSION TITLE
Representations from representation_models are generated twice when nr_topics="auto". Is this expected?

### DISCUSSION THREAD

**Discussion by siddharthtumre** [MaartenGr/BERTopic#2142] [Q&A]
Hi @MaartenGr 

While using nr_topics="auto", I can see that the representations from representation_models are calculated after reducing the number of topics. This increases the run time as a LLM is prompted twice i.e., before and after reducing topics.

I believe generating representations after reducing the topics should be sufficient.

**Comment by MaartenGr** [comment]
That's indeed what is happening currently. I believe it should be straightforward to implement that so this might be a good first issue for those wanting to help out. 

For now, what you could also do is run the model first without any representation models and then use `.update_topics` without your selected representation model (e.g, LLM) to create the representations. Then the LLM would run only once and you get the same results.

**Reply by siddharthtumre** [reply]
Thanks for the response.

I will try to update the code and let you know.
